### ISSUE TITLE
Getting "undefined" error when running validations imperatively with "oneOf"

### ISSUE THREAD

**Issue by armandAkop** [express-validator/express-validator#995]
## Describe the bug

I am running a simple validation using oneOf and running the validations imperatively. My validation middleware calls validation.run(validation), the result is undefined.

## To Reproduce

**The validation middleware taken from the docs**
```node
const requestValidator = (validations = []) => {
  return async (req, res, next) => {
    for (const validation of validations) {
      const result = await validation.run(req);

      if (result.errors.length) break; // TypeError: Cannot read property 'errors' of undefined
    }

    const errors = validationResult(req);
    if (errors.isEmpty()) {
      return next();
    }
    
    throw new ValidationError(errors.errors[0].msg);
  }
};
```

**Sample route that is failing**
```node
const validationsToRun = [
  oneOf([
    body('ipAddress').isIP()
  ])
];

router.post('/test', requestValidator(validationsToRun), (req, res, next) => {
  res.send('sall good');
});
```

**Payload**
```JSON
{
    "ipAddress": 4.4
}
```
## Expected behavior

I expect the imperative validation would work as is, given that function comes from the documentation. And it seems like [this issue ](https://github.com/express-validator/express-validator/issues/822) was similar,  but was resolved with a PR.

## Express-validator version:

- Version: 6.10.0

Am I doing something wrong here?

**Comment by fedeci** [comment]
Does it work correctly if you don't run it imperatively?

**Comment by armandAkop** [comment]
Awesome, thanks!

**Comment by fedeci** [comment]
We actually have to figure out what result we want to expose. Furthermore some of those properties are marked as `private` (which i just a `tsc` flag) and I think they should be marked as private, in v7.0.0 because this is breaking, also for the end user using `#propname`.
/cc @gustavohenke

**Comment by gustavohenke** [comment]
Hey there! This is fixed on v6.11.0 🚀
